### PR TITLE
Integration test for Mesos checks for apps

### PR DESCRIPTION
**Pull request by kensipe**
Summary:
Added integration tests for Mesos Checks for apps.   Currently checking command checks for the pass and fail conditions.

JIRA issues: DCOS-51842

### CODE REVIEW

File: tests/integration/src/test/scala/mesosphere/marathon/integration/facades/MesosFacade.scala [hunk 1]
```diff
@@ -116,7 +116,7 @@ class MesosFacade(val url: String, val waitTime: FiniteDuration = 30.seconds)(im
 
   def state: RestResult[ITMesosState] = {
     logger.info(s"fetching state from $url")
-    result(requestFor[ITMesosState](Get(s"$url/state.json")), waitTime)
+    result(requestFor[ITMesosState](Get(s"$url/state")), waitTime)
```
Comment (kensipe) [inline tests/integration/src/test/scala/mesosphere/marathon/integration/facades/MesosFacade.scala]: we need to talk about this...   `/state` is required now with Mesos 1.8.   It was introduced with an earlier version of mesos.   Do we require all devs to switch to a version of mesos that works for `/state` or do we detect and switch from `/state.json` and `/state` dependent on version?

Comment (zen-dog) [reply tests/integration/src/test/scala/mesosphere/marathon/integration/facades/MesosFacade.scala]: `/state` is supported in older Mesos versions so we can just leave it like this

File: tests/integration/src/test/scala/mesosphere/marathon/integration/MesosCheckIntegrationTest.scala [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+import java.util.UUID
+
+import mesosphere.AkkaIntegrationTest
+import mesosphere.marathon.integration.facades.MarathonFacade.extractDeploymentIds
+import mesosphere.marathon.integration.setup.EmbeddedMarathonTest
+import mesosphere.marathon.raml.{App, AppCheck, CommandCheck, ShellCommand}
+import mesosphere.marathon.state.PathId
+
+class MesosCheckIntegrationTest extends AkkaIntegrationTest with EmbeddedMarathonTest {
```
Comment (timcharper) [inline tests/integration/src/test/scala/mesosphere/marathon/integration/MesosCheckIntegrationTest.scala]: Maybe we can combine this with `HealthCheckIntegrationTest` ? Rename to "CheckIntegrationTest" ? There's a startup / teardown cost for each suite... seems like maybe we should optimize for this if we can?

File: tests/integration/src/test/scala/mesosphere/marathon/integration/MesosCheckIntegrationTest.scala [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        cmd = Some(appCommand),
+        check = Some(ramlCommandCheck("true")))
+
+      When("The app deploys")
+      val result = marathon.createAppV2(app)
+
+      Then("The app is created with a successful exit code")
+      result should be(Created)
+      extractDeploymentIds(result) should have size 1
+      waitForDeployment(result)
```
Comment (timcharper) [inline tests/integration/src/test/scala/mesosphere/marathon/integration/MesosCheckIntegrationTest.scala]: Feels like we should have another `Then` here? 

` Then("The app is created")`

`And("the successful exit code is exposed via tasks")`

File: tests/integration/src/test/scala/mesosphere/marathon/integration/MesosCheckIntegrationTest.scala [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+      eventually {
+        val currentTasks = marathon.tasks(id).value
+        currentTasks should have size (1)
+        currentTasks.head.check.get.command.get.exitCode should be (0)
+      }
+      marathon.deleteApp(id)
```
Comment (timcharper) [inline tests/integration/src/test/scala/mesosphere/marathon/integration/MesosCheckIntegrationTest.scala]: FYI - this happens automatically on teardown :)

File: tests/integration/src/test/scala/mesosphere/marathon/integration/MesosCheckIntegrationTest.scala [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+
+    "provide a command check result status of 1" in {
+      Given("a new app")
+      val id = appId(Some("with-failing-checks"))
+      val app = App(
+        id.toString,
+        cmd = Some(appCommand),
+        check = Some(ramlCommandCheck("false")))
+
+      When("The app deploys")
```
Comment (timcharper) [inline tests/integration/src/test/scala/mesosphere/marathon/integration/MesosCheckIntegrationTest.scala]:
```suggestion
      When("the app is created")
```

File: tests/integration/src/test/scala/mesosphere/marathon/integration/MesosCheckIntegrationTest.scala [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+      val result = marathon.createAppV2(app)
+
+      Then("The app is created with a failing exit code")
```
Comment (timcharper) [inline tests/integration/src/test/scala/mesosphere/marathon/integration/MesosCheckIntegrationTest.scala]:
```suggestion
      Then("the app deploys successfully")
```

File: tests/integration/src/test/scala/mesosphere/marathon/integration/MesosCheckIntegrationTest.scala [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+      result should be(Created)
+      extractDeploymentIds(result) should have size 1
+      waitForDeployment(result)
+      eventually {
```
Comment (timcharper) [inline tests/integration/src/test/scala/mesosphere/marathon/integration/MesosCheckIntegrationTest.scala]:
```suggestion
      And("the faililng check status is exposed in /v2/tasks")
      eventually {
```

**Pull request merged**